### PR TITLE
Feature/jlin/cns 36 upload file to s3 function

### PR DESCRIPTION
**Pull request by JLin056**
## Change
- Add upload file to s3 function
- Add S3-related tests:
   - Test file: `test.txt`
   - add upload file test: `S3Test.js`
 
### About test
The test simulates uploading a file to S3 to verify that the S3Service correctly handles file uploads.

### How to run the test
1. Log in to AWS IAM (S3UploaderDownloader) and create access key
2. Add the generated access key and secret key to your `.env` file:
```
AWS_ACCESS_KEY_ID=your-access-key-id
AWS_SECRET_ACCESS_KEY=your-secret-access-key
S3_BUCKET_REGION=your-bucket-region
BUCKET_NAME=your-bucket-name
```
3. Navigate to the project directory and install dependencies
```
cd backend
npm install
```
4. Execute the test script:
```
node src/tests/s3/s3Test.js
```

### Expected results
If the file upload is successful, the result will include the message:
```
Upload Success: {file information}
```

For Example:
image

### CODE REVIEW

File: backend/src/services/s3Service.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+        const key = this._generateS3Key(type, id, filename); // S3 file key
+
+        // 驗證檔案是否存在
+        if (!fs.existsSync(file.tempFilePath)) {
+            throw new Error(`[S3Service] File not found: ${file.tempFilePath}`);
+        }
+
+        // 檔案流的錯誤
+        const fileStream = fs.createReadStream(file.tempFilePath);
```
Comment (yun-si) [inline backend/src/services/s3Service.js:40]: 有點不太懂這個註解是什麼意思 😂

File: backend/test/s3Test.js
Comment (yun-si) [inline backend/test/s3Test.js:1]: 可以改一下檔名🥹
s3.integration.test

**Review by yun-si — CHANGES_REQUESTED** [review]

File: backend/src/services/s3Service.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import { S3Client, PutObjectCommand } from "@aws-sdk/client-s3";
+import fs from "fs";
+import path from "path";
```
Comment (yun-si) [inline backend/src/services/s3Service.js]: 這個其實沒有用到，可以拿掉～

File: backend/src/services/s3Service.js (hunk 1 above)
Comment (JLin056) [reply backend/src/services/s3Service.js:40]: 那時候改程式碼，忘記刪掉這個註解了😅我會刪掉！

**Pull request closed without merging**